src/tiramisu_core.cpp

```diff
@@ -7118,10 +7162,12 @@ void tiramisu::buffer::tag_gpu_shared() {
 
 void tiramisu::buffer::tag_gpu_constant() {
     location = cuda_ast::memory_location::constant;
+    this->auto_trans = false ;    
 }
 
 void tiramisu::buffer::tag_gpu_global() {
     location = cuda_ast::memory_location::global;
```

Comment:
Use this mechanism for specifying that a buffer is in constant memory or for figuring out whether a buffer should be in global memory.

Reply:
I'm not talking about that. I'm talking about the fact that you created a new variable that stores whether the buffer is global or constant. We already had the "location" that indicates whether the buffer is global/constant, ... and you added a new member "tag" that has exactly the same role. I meant you should use "location" instead of introducing "tag".

Regarding the introduction of a new input for the buffer constructor to specify whether the buffer is global, I also do not agree with that. Your new code should be compatible with the existing API. In the existing API, we declare a buffer and then specify where it should be stored using a function like "tag_gpu_constant()", I prefer if we keep that instead of introducing a new input for the buffer constructor. The only reason for this is to keep compatibility with the existing API. Once people learn the API, they expect the whole API to behave in the same way. SO I prefer if you remove that too.